lms/templates/discussion/_discussion_inline.html

```diff
@@ -6,13 +6,17 @@
 from openedx.core.djangolib.js_utils import js_escaped_string
 %>
 
-<div class="discussion-module" data-discussion-id="${discussion_id}" data-user-create-comment="${json_dumps(can_create_comment)}" data-user-create-subcomment="${json_dumps(can_create_subcomment)}" data-read-only="false">
-    <a class="discussion-show control-button" href="javascript:void(0)" data-discussion-id="${discussion_id}" role="button">
-        <span class="show-hide-discussion-icon"></span><span class="button-text">${_("Show Discussion")}</span>
-    </a>
-    % if can_create_thread:
-        <button class="btn-brand btn-small new-post-btn">${_("Add a Post")}</button>
-    % endif
+<div class="discussion-module" data-discussion-id="${discussion_id}"
+     data-user-create-comment="${json_dumps(can_create_comment)}"
+     data-user-create-subcomment="${json_dumps(can_create_subcomment)}"
+     data-read-only="${'false' if can_create_thread else 'true'}">
+    <div class="discussion-module-title">
+        <h3 class="h3--discussion-module">${_(display_name)}</h3>
+        <div class="inline-discussion-topic"><strong>${_("Topic:")}</strong> ${discussion_category} / ${discussion_target}</div>
+    </div>
+    <button class="discussion-show btn btn-brand" data-discussion-id="${discussion_id}">
```

Comment:
I don't think you need both "btn" and "btn-brand". At least, that's true for the Pattern Library. You might also want "btn-small". See the "Add a Post" button element you removed above.